### PR TITLE
feat: add support to palm version

### PR DESCRIPTION
**Pull request by Henrrypg**
## Description
Support tutor 16.

## What has changed from Tutor15?

- Tutor variables RUN_CMS and RUN_LMS [were deprecated](https://github.com/overhangio/tutor/commit/ee8de62770c6d6ba624805347acc7d4d30dfd0a0).
- The way to access to drydock context [changed](https://github.com/overhangio/tutor/blob/74ef1ae56e9bbd4c6a9441a10bb90576136243be/CHANGELOG.md?plain=1#L41).

## Testing Instructions.

Using tutor V16.0.2 install used plugins ([tutor-notes](https://github.com/overhangio/tutor-notes), [tutor-forum](https://github.com/overhangio/tutor-forum), [tutor-minio](https://github.com/overhangio/tutor-minio)) and create the manifest files.

`tutor drydock save -r drydock/drydock/references/tutor_v16.yml`

### CODE REVIEW

**Comment by MoisesGSalas** [comment]
@Henrrypg, did you deploy manage to deploy a new live environment with the palm version?

**Comment by Henrrypg** [comment]
> @Henrrypg, did you deploy manage to deploy a new live environment with the palm version?

@MoisesGSalas I have some changes in yamato with a new environment using this version, i'll push and deploy

File: drydock/templates/kustomized/tutor16/base/apps/openedx/config/cms.env.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+SITE_NAME: "{{ CMS_HOST }}"
```
Comment (jfavellar90) [inline drydock/templates/kustomized/tutor16/base/apps/openedx/config/cms.env.yml:1]: Please include the uwsgi template from [tutor](https://github.com/overhangio/tutor/blob/master/tutor/templates/apps/openedx/uwsgi.ini)

File: drydock/templates/kustomized/tutor16/base/apps/openedx/config/cms.env.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+BOOK_URL: ""
+LOG_DIR: "/openedx/data/logs"
```
Comment (jfavellar90) [inline drydock/templates/kustomized/tutor16/base/apps/openedx/config/cms.env.yml:3]: @Henrrypg please make sure there are no missing templates

File: drydock/templates/kustomized/tutor16/base/apps/openedx/settings/partials/common_all.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+# Mongodb connection parameters: simply modify `mongodb_parameters` to affect all connections to MongoDb.
+mongodb_parameters = {
+    "db": "{{ MONGODB_DATABASE }}",
+    "host": "{{ MONGODB_HOST }}",
+    "port": {{ MONGODB_PORT }},
+    "user": {% if MONGODB_USERNAME %}"{{ MONGODB_USERNAME }}"{% else %}None{% endif %},
+    "password": {% if MONGODB_PASSWORD %}"{{ MONGODB_PASSWORD }}"{% else %}None{% endif %},
+    # Connection/Authentication
+    "ssl": {{ MONGODB_USE_SSL }},
+    "authSource": "{{ MONGODB_AUTH_SOURCE }}",
```
Comment (jfavellar90) [inline drydock/templates/kustomized/tutor16/base/apps/openedx/settings/partials/common_all.py:16]:
```suggestion
    "authsource": "{{ MONGODB_AUTH_SOURCE }}",
```
From https://github.com/overhangio/tutor/pull/841

File: drydock/templates/kustomized/tutor16/base/apps/openedx/settings/partials/common_lms.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+{% include "apps/openedx/settings/partials/common_all.py" %}
+
+######## Common LMS settings
+LOGIN_REDIRECT_WHITELIST = ["{{ CMS_HOST }}"]
+
+# Better layout of honor code/tos links during registration
+REGISTRATION_EXTRA_FIELDS["terms_of_service"] = "required"
```
Comment (jfavellar90) [inline drydock/templates/kustomized/tutor16/base/apps/openedx/settings/partials/common_lms.py:7]:
```suggestion
REGISTRATION_EXTRA_FIELDS["terms_of_service"] = "hidden"
```
From https://github.com/overhangio/tutor/commit/2d6b9c17674ebcc1ebd8cc4f56c3b0413c171300

File: drydock/templates/kustomized/tutor16/extensions/hpa.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
+---
+apiVersion: autoscaling/v1
```
Comment (jfavellar90) [inline drydock/templates/kustomized/tutor16/extensions/hpa.yml:2]: Should we move to the V2 of the [HPA api](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#horizontalpodautoscaler-v2-autoscaling)?

Comment (Henrrypg) [reply drydock/templates/kustomized/tutor16/extensions/hpa.yml:2]: Currently V2 is failing due to https://github.com/argoproj/argo-cd/issues/9145 in out ArgoCD version, it shows an error when is deployed and shows unknow status. In ArgoCD Stage is working fine. @jfavellar90

Comment (jfavellar90) [reply drydock/templates/kustomized/tutor16/extensions/hpa.yml:2]: To prevent blocking this, just move forward with V1

**Comment by jfavellar90** [comment]
@Henrrypg I don't have additional in terms of code. I checked Nagasaki and the environment was initialized OK in Argo and also checked the installation is functional. I give thumbs up

**Review by jfavellar90 — APPROVED** [review]

**Review by MoisesGSalas — APPROVED** [review]

**Pull request merged**